src/network/networkd-link.c

```diff
@@ -1972,8 +1982,14 @@ static int link_joined(Link *link) {
                 r = link_set_bridge(link);
                 if (r < 0)
                         log_link_error_errno(link, r, "Could not set bridge message: %m");
+
         }
 
+        /* TODO: check for bridge interface? */
```

Comment:
You are right, currently the NetDev and the Link is entirely disconnected. The philosophy so far being that any low-level (i.e., link-type specific) configuration of a NetDev we create ourselves should be applied at creation (so set in the .netdev file and passed in with NEWLINK), and low-level configuration applied to Links should only be done to override kernel defaults for links not created by userspace.

It is sort of nice that all properties are correct from the get-go, but of course, if this is not possible, and we need to adjust some settings after the bridge has been created, we should do that. Either by linking to the netdev, or probably better just track the link type in the Link object.